app/questionnaire/questionnaire_schema.py

```diff
@@ -336,12 +343,32 @@ def _update_answer_dependencies_for_value_source(
                     self._answer_dependencies_map[answer_id_for_block] |= {
                         self._get_answer_dependent_for_block_id(block_id=block_id, answer_id=answer_id)  # type: ignore
                     }
+        if value_source["source"] == "list":
+            section = self.get_section_for_block_id(block_id)
+            list_collector = self.get_list_collector_for_list(
+                section=section, for_list=value_source["identifier"]  # type: ignore
+            )
+            # Returns optional, but it always returns at this point, same with optional returns below
+            add_block_question = self.get_add_block_for_list_collector(  # type: ignore
+                list_collector["id"]  # type: ignore
+            )["question"]
+            answer_id_for_block = list(
+                self.get_answers_for_question_by_id(add_block_question).keys()
+            )[0]
```

Comment:
We fetch first here because that method (`get_answers_for_question_by_id`) returns a list even when there is only one element on it. Answer dependencies for list source were not populated before hence this change. I had to add the whole list source logic from scratch.